dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java

```diff
@@ -914,6 +914,13 @@ private void readDataInFetch()
             try
             {
                 bytesToRead = this.dis.available();
+
+                // Either due to a bug in the JVM or due to a design issue
+                // available can return 0 even in cases where there is data available.
+                // So always attempt to read at least some data
+                if (bytesToRead < MIN_SOCKET_READ_SIZE) {
```

Comment:
There seems to be many questions about his online 
(https://stackoverflow.com/questions/11982693/available-of-datainputstream-from-socket
https://stackoverflow.com/questions/26527918/datainputstream-available-returns-0-when-there-should-be-data-available)
It seems reading even if no data available is the accepted answer. 
I wonder if L921 should ensure available > 0 ( I think -1 is possible and signifies end of stream)

Also, curly brace should be in new line.